datafusion/optimizer/Cargo.toml

```diff
@@ -44,6 +44,7 @@ async-trait = { workspace = true }
 chrono = { workspace = true }
 datafusion-common = { workspace = true }
 datafusion-expr = { workspace = true }
+datafusion-functions = { workspace = true }
```

Comment:
Is it only used in testing? Then it should be placed under `dev-dependencies`.

Reply:
Good catch!

Reply:
FYI @Omega359  had a similar challenge when looking at timestamp functions. We are thinking it might be better to move the tests rather than add a new dependency -- see https://github.com/apache/arrow-datafusion/issues/9291#issuecomment-1957832283

Reply:
I don't think we can add this as a dependency as it prevents the crates from being published to crates.io. See https://github.com/apache/arrow-datafusion/issues/9277

I think we should move these tests out of the optimizer crate

For this PR, however, perhaps we could simply update this test to use a function that has not yet been ported and maybe move the tests in another PR

Reply:
> For this PR, however, perhaps we could simply update this test to use a function that has not yet been ported and maybe move the tests in another PR

done